emloop/tests/utils/config_test.py

```diff
@@ -65,3 +65,96 @@ def test_load_anchored_config(tmpdir, anchored_yaml, params, a_key, e_key):
 
     assert load_config(f_name, *params)['a'] == a_key
     assert OrderedDict(load_config(f_name, *params)['e']) == OrderedDict(e_key)
+
+
+@pytest.fixture
+def yaml():
+    yield """
+          model:
+            class: Model
+
+            io: 
+              in: [a]
+              out: [b]
+
+            outputs: [c]
+
+          dataset:
+            class: Dataset
+            
+            batch_size: 10
+
+          hooks:
+          - Hook_1:
+              epochs: 1
+
+          - Hook_2:
+              epochs: 1
+
+          eval:
+            train:      
+              model:
+                class: EvalModel
+
+                io: 
+                  in: [x]
+                  out: [y]
+                  
+                inputs: [z]
+
+              dataset:
+                class: EvalDataset
+
+              hooks:
+              - Hook_2:
+                  epochs: 2
+          """
+
+
+def test_override(tmpdir, yaml):
+    """Test configuration is overridden by the eval section and subsequently by the CLI arguments."""
+    orig_config = path.join(tmpdir, 'test.yaml')
+
+    with open(orig_config, 'w') as file:
+        file.write(yaml)
+
+    cl_arguments = ['dataset.class=CliDataset', 'model.io.in=[m]']
+
+    # neither cl_arguments nor override_stream specified - no override
+    config_0 = {'model': {'class': 'Model', 'io': {'in': ['a'], 'out': ['b']}, 'outputs': ['c']},
```

Comment:
You dont need to specify the whole dict (it is hard to find out what is actually tested). Just test the respective sections in the config.